charts/shoot-core/components/charts/apiserver-proxy/templates/apiserver-proxy-daemonset.yaml

```
@@ -0,0 +1,105 @@
+---
+apiVersion: {{ include "daemonsetversion" . }}
+kind: DaemonSet
+metadata:
+  name: apiserver-proxy
+  namespace: kube-system
+  labels:
+    garden.sapcloud.io/role: system-component
```

For newly added components it does not make sense to use the deprecated key.
```suggestion
    gardener.cloud/role: system-component
```